server/sonar-ce/src/test/java/org/sonar/ce/container/ComputeEngineContainerImplTest.java

```diff
@@ -63,19 +67,63 @@ public void constructor_does_not_create_container() {
   }
 
   @Test
-  public void real_start() throws IOException {
-    Properties properties = ProcessProperties.defaults();
-    File homeDir = tempFolder.newFolder();
-    File dataDir = new File(homeDir, "data");
-    File tmpDir = new File(homeDir, "tmp");
-    properties.setProperty(PATH_HOME, homeDir.getAbsolutePath());
-    properties.setProperty(PATH_DATA, dataDir.getAbsolutePath());
-    properties.setProperty(PATH_TEMP, tmpDir.getAbsolutePath());
-    properties.setProperty(PROPERTY_PROCESS_INDEX, valueOf(ProcessId.COMPUTE_ENGINE.getIpcIndex()));
-    properties.setProperty(PROPERTY_SHARED_PATH, tmpDir.getAbsolutePath());
-    properties.setProperty(DatabaseProperties.PROP_URL, ((BasicDataSource) dbTester.database().getDataSource()).getUrl());
-    properties.setProperty(DatabaseProperties.PROP_USER, "sonar");
-    properties.setProperty(DatabaseProperties.PROP_PASSWORD, "sonar");
+  public void real_start_with_cluster() throws IOException {
+    int port = NetworkUtils.freePort();
+    HazelcastInstance hzInstance = HazelcastTestHelper.createHazelcastCluster(CLUSTER_NAME, port);
+
+    Properties properties = getProperties();
+    properties.setProperty(ProcessProperties.CLUSTER_ENABLED, "true");
+    properties.setProperty(ProcessProperties.CLUSTER_LOCALENDPOINT, String.format("%s:%d", hzInstance.getCluster().getLocalMember().getAddress().getHost(), port));
+    properties.setProperty(ProcessProperties.CLUSTER_NAME, CLUSTER_NAME);
+
+    // required persisted properties
+    insertProperty(CoreProperties.SERVER_ID, "a_startup_id");
+    insertProperty(CoreProperties.SERVER_STARTTIME, DateUtils.formatDateTime(new Date()));
+
+    underTest
+      .start(new Props(properties));
+
+    MutablePicoContainer picoContainer = underTest.getComponentContainer().getPicoContainer();
```

Comment:
don't fork all this assertions on pico containers, it's going to be a pain to maintain.

Either make you test more specific (I guess you wanted to check classes `ClusterClient` and `ClusterProperties` are added, so just test that and rename the class to describe exactly that) or factor the testing code.

I favor the first solution which you can complete by added an assertion to `real_start_without_cluster` which ensure the classes aren't there.